From 195e2d9febfbeef1d09701c387925e5c2f5cb038 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C5=82=20Mi=C5=82ecki?= <rafal@milecki.pl>
Date: Thu, 11 Feb 2021 13:12:38 +0100
Subject: [PATCH] net: broadcom: bcm4908_enet: fix received skb length
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use ETH_FCS_LEN instead of magic value and drop incorrect + 2

Signed-off-by: Rafał Miłecki <rafal@milecki.pl>
Reviewed-by: Florian Fainelli <f.fainelli@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/broadcom/bcm4908_enet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/broadcom/bcm4908_enet.c
+++ b/drivers/net/ethernet/broadcom/bcm4908_enet.c
@@ -567,7 +567,7 @@ static int bcm4908_enet_poll(struct napi
 
 		dma_unmap_single(dev, slot.dma_addr, slot.len, DMA_FROM_DEVICE);
 
-		skb_put(slot.skb, len - 4 + 2);
+		skb_put(slot.skb, len - ETH_FCS_LEN);
 		slot.skb->protocol = eth_type_trans(slot.skb, enet->netdev);
 		netif_receive_skb(slot.skb);
 
